app/connectors/cache/MongoCacheConnector.scala

```diff
@@ -92,9 +152,12 @@ class MongoCacheConnector @Inject()(cacheClientFactory: MongoCacheClientFactory)
     * @return The model after it has been transformed
     */
   def update[T](key: String)(f: Option[T] => T)(implicit ac: AuthContext, hc: HeaderCarrier, fmt: Format[T]): Future[Option[T]] =
-    mongoCache.find[T](ac.user.oid, key) flatMap { maybeModel =>
-      val transformed = f(maybeModel)
-      mongoCache.createOrUpdate(ac.user.oid, transformed, key) map { _ => Some(transformed) }
+    authConnector.getCredId flatMap {
+      credId =>
+        mongoCache.findWithCacheMiss[T](ac.user.oid, credId, key) flatMap { maybeModel =>
+          val transformed = f(maybeModel)
```

Comment:
Looking at what f does, I have a bad feeling about it

Reply:
This line could possibly be removed

Reply:
Would `mongoCache.createOrUpdate(credId, ac.user.oid, transformed, key) map { _ => maybeModel }` work?

Reply:
Not sure, for me looks good: find -> transform using f -> save, and is used in few places in service